modules/devices.py

```diff
@@ -108,24 +99,49 @@ def autocast(disable=False):
 
 # MPS workaround for https://github.com/pytorch/pytorch/issues/79383
 orig_tensor_to = torch.Tensor.to
+
+
 def tensor_to_fix(self, *args, **kwargs):
-    if self.device.type != 'mps' and \
-       ((len(args) > 0 and isinstance(args[0], torch.device) and args[0].type == 'mps') or \
-       (isinstance(kwargs.get('device'), torch.device) and kwargs['device'].type == 'mps')):
+    if self.device.type != 'mps' and (
+        args
+        and isinstance(args[0], torch.device)
+        and args[0].type == 'mps'
+        or (
+            isinstance(kwargs.get('device'), torch.device)
+            and kwargs['device'].type == 'mps'
+        )
+    ):
         self = self.contiguous()
     return orig_tensor_to(self, *args, **kwargs)
 
 
-# MPS workaround for https://github.com/pytorch/pytorch/issues/80800 
+# MPS workaround for https://github.com/pytorch/pytorch/issues/80800
 orig_layer_norm = torch.nn.functional.layer_norm
+
+
 def layer_norm_fix(*args, **kwargs):
-    if len(args) > 0 and isinstance(args[0], torch.Tensor) and args[0].device.type == 'mps':
+    if (
+        args
+        and isinstance(args[0], torch.Tensor)
+        and args[0].device.type == 'mps'
+    ):
         args = list(args)
         args[0] = args[0].contiguous()
     return orig_layer_norm(*args, **kwargs)
 
 
+# MPS workaround for https://github.com/pytorch/pytorch/issues/90532
+orig_tensor_numpy = torch.Tensor.numpy
+
+
+def numpy_fix(self, *args, **kwargs):
+    if self.requires_grad:
+        self = self.detach()
+    return orig_tensor_numpy(self, *args, **kwargs)
+
+
```

Comment:
This added method and then using it on line 147 made it so I could save my embedding checkpoints rather than crash.